### ISSUE TITLE
Provide automated checks for code styling & conventions

### ISSUE THREAD

**Issue by tomkerkhove** [arcus-azure/arcus.eventgrid#39]
Provide automated checks for code styling & conventions so that the codebase is consistent.

A new person contributing should be up & running as quickly as possible.

## Proposal
This should be well-documented and enforced on the CI

Options are:
- Resharper Code Quality
- .NET / VS tooling
- StyleCop

**Comment by tomkerkhove** [comment]
Relates to https://github.com/arcus-azure/arcus/issues/12

**Comment by stijnmoreels** [comment]
I would like this. But maybe choose an option with the least installation and configuration option.
Also we should take into account that every dev has a different opinion on code style and it maybe not a good idea to make the enforcement too strict.

Off course, we should verify the logical ones: read-only, disposal, unused, await, public doc, null, ...

**Comment by tomkerkhove** [comment]
That's correct but we have to set a baseline so that the codebase is consistent and people know what to expect/deliver.

Styling is a hard topic so there will always be rules which not everybody agrees upon, but that's how it will be.

There should indeed be no installation or configuration, just pull the codebase and ready to roll!
